### ISSUE TITLE
Add clarification regarding event listener in event batching section

### ISSUE THREAD

**Issue by mikep-dev** [GoogleChrome/web-vitals#502]
I believe the section regarding event batching requires some clarification.
https://github.com/GoogleChrome/web-vitals?tab=readme-ov-file#batch-multiple-reports-together

The provided code works fine, but I feel it's worth emphasizing the importance of attaching listener for the `visibilitychange` event to the correct target. Check out this demo:
https://stackblitz.com/edit/stackblitz-starters-faxsov?file=index.html

In the demo I've attached event listener to `document` and `window` and configured `onCLS` to add metric to queue and log message to console.

![image](https://github.com/GoogleChrome/web-vitals/assets/15386741/e9c4be5a-3d4d-4fa8-b9e1-a16134166414)

As you can see, if you were to use `document.addEventListener` instead of global `addEventListener` or `window.addEventListener` you would lose CLS data, since CLS metric is reported after that. 

This happens because `onCLS` reports data on `visibilitychange` on `document` but it attaches its listener inside `onFCP` callback. That means our listener that sends analytics data will be attached (and executed) earlier if its attached to the same target. The solution is simple: take advantage of event bubbling and attach the listener to `window` so that it executes after `onCLS` internal listener.

I've made a mistake and targeted `document` and wasted time scratching my head wondering why all other metrics are collected on time, but CLS was always late. I believe adding a proper note to `README` might save some developers time in the future, for example:

> ***Note**: to make sure your event listener is executed after all metrics have been reported, attach it to `window` object with `addEventListener` or `window.addEventListener` instead of `document.addEventListener`, which is internally used by some reporters.*

**Comment by philipwalton** [comment]
Yes, we should update this library to ensure that if someone calls any of the various `onXXX()` functions, the moment that happens all necessary `visibilitychange` listeners get added. That way as long as folks call the `onXXX()` functions before registering their own listeners, our listeners will run first.

We already have the `getVisibilityWatcher.js` module that is initialized early by most of these metric functions. Why don't we add an `onHidden()` method to the [returned watcher object](https://github.com/GoogleChrome/web-vitals/blob/15a377360f3ee7303ba68edc579b1c725111bd98/src/lib/getVisibilityWatcher.ts#L96-L100) that accepts a callback, and then all metric functions can plug into that rather than adding their own `visibilitychange` listeners later in the lifecycle. WDYT?

**Comment by tunetheweb** [comment]
Sounds good. There’s also a few cases (including that one!) that register the `visibilitychange` event later without the `document.` prefix so should fix that at the same time.
